docs/how-to/push-update-on-mainnet.md

```diff
@@ -0,0 +1,192 @@
+* `0` is new script version. Again, it's the same as the current one.
+* `20` is new slot duration (in seconds). Unchanged.
+* `2000000` is new maximum block size (in bytes). Unchanged.
+
+The next argument (`csl-daedalus:1`) is software version description. You should substitute `1` (version) with the integer provided along with installers (see *Prerequisites* section, *4.* item)
+
+Next follow an arbitrary number of _triples_ (two in our case, one for Windows and one for macOS). Each triple is `<system tag> <installer filename> none`. Once again, those are filenames in the current dir, not arbitrary paths. `none` stands for binary diff (this feature is not used for now).
+
+Successfull command output looks like this:
+
+```
+[smart-wallet:DEBUG] [2017-09-21 15:13:24 MSK] Proposing update...
+Read file daedalus-win64-1.0.3350.0-installer.exe succesfuly, its hash: 01abf1c8b881c2f8ea4d1349a700f29d4088e68dc04b6bf4680ea7e14638373e
+Read file installer062macos64.pkg succesfuly, its hash: 3bc1084841fb99fff03ef92bc35eef9a80a20aeca688dfc1b50a4aa6dd6f7c73
+[smart-wallet:INFO] [2017-09-21 15:13:25 MSK] Announcing proposal with id 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6
+Update proposal submitted, upId: 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6
+```
+
+Note:
+
+```
+Read file daedalus-win64-1.0.3350.0-installer.exe succesfuly, its hash: 01abf1c8b881c2f8ea4d1349a700f29d4088e68dc04b6bf4680ea7e14638373e
+Read file Daedalus-installer-1.0-rc.3202.pkg succesfuly, its hash: 3bc1084841fb99fff03ef92bc35eef9a80a20aeca688dfc1b50a4aa6dd6f7c73
+```
+
+These hashes should be used to form URLs for installers on S3
+buckets. In this example the URL `<update
+server>/01abf1c8b881c2f8ea4d1349a700f29d4088e68dc04b6bf4680ea7e14638373e`
+should respond with the contents of `daedalus-win64-1.0.3350.0-installer.exe` and the URL
+`<update
+server>/3bc1084841fb99fff03ef92bc35eef9a80a20aeca688dfc1b50a4aa6dd6f7c73`
+should respond with the contents of `Daedalus-installer-1.0-rc.3202.pkg`.
+
+These hashes are Blake2b_256 hashes of CBOR-encoded contents of the files.
+There is a simple command to calculate the hash of an installer:
+
+```
+stack exec -- cardano-auxx $COMMONOPTS $AUXXOPTS cmd --commands "hash-installer <FILEPATH>"
+```
+
+Note:
+
+```
+Update proposal submitted, upId: 4c827d6fe03d4c3646ebbbc28d4e09c57690e1dcba54b9adc0050d3f76734cf6
+```
+
+Value `upId` is used on one of next steps, when we'll vote for update. You may also look the first six characters of the hash on Papertrail to see whether the proposal was accepted by nodes or not.
+
+Uploading update files
+======================
+
+Upload installers to S3 bucket:
+* For staging:
+    * URL: https://s3.eu-central-1.amazonaws.com/update-system-testing/
+    * S3 Bucket: `update-system-testing`
+* For mainnet:
+    * URL: https://update.cardano-mainnet.iohk.io
```

Comment:
the URL in https://github.com/input-output-hk/daedalus/blob/eb713a66eb2c0445fbe8c2faa59f0884edd83712/installers/Launcher.hs#L68 does not use TLS
and the SSL cert for that domain is currently invalid